### ISSUE TITLE
FEATURE REQUEST: mapping via static call in-Editor

### ISSUE THREAD

**Issue by CaseyHofland** [starikcetin/Eflatun.SceneReference#93]
I'm a big fan of SceneReference but I never liked how it created json files for serialization inside a folder you need to source control. It keeps me from referencing this package inside of other packages, since I would have to communicate users they have to put these extra lines in their source control, and users may find my packages from multiple different sources.

What if you serialized guids inside a json file only once you build the game? And then, in the editor, it uses (something like) this?

```csharp
public class SceneReference
{
    public SceneAsset sceneAsset;

    public string path => AssetDatabase.GetAssetPath(sceneAsset);
}
```

Obviously wrapped in `#if UNITY_EDITOR` where appropriate.

**Comment by starikcetin** [comment]
I see where you are coming from. Let me think for a while about this one.

**Comment by starikcetin** [comment]
Update: I managed to get it working, I am considering whether to merge it or not.

**Comment by CaseyHofland** [comment]
That's fantastic! Happy to hear this was backwards compatible, I was worried about that.

**Comment by starikcetin** [comment]
Even though it requires no public API changes, I will still do a major bump for this one, as it is a fundamental change in behaviour.

I have been considering ditching the JSON generation completely and instead going with [source generators](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) for a while. If we go that route, then this issue would be implicitly solved.

Therefore it is going to take a while for me to thoroughly decide on this one.

**Comment by CaseyHofland** [comment]
That would be infinitely better!

By all means, take your time, you clearly know what you're doing. If you push this feature branch then I will fork it until v4 hits.

**Comment by starikcetin** [comment]
Update: It doesn't look possible to reliably use source generators for this library. Therefore I am abandoning that idea. I am back to evaluating the original idea.

**Comment by starikcetin** [comment]
Everyone interested, please review PR #94.

**Comment by starikcetin** [comment]
Implemented in #94. Will be included in the next release.

**Comment by starikcetin** [comment]
Included in [4.0.0](https://github.com/starikcetin/Eflatun.SceneReference/releases/tag/4.0.0).
